### PR TITLE
Internal: Fix prop deps [ED-23408]

### PR DESCRIPTION
**Pull request by marcin-el**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

## 1. Problem & Context

Property dependency restoration was broken when overridable wrappers (runtime value overrides) existed. The code didn't account for shape mismatches between saved state and current prop structure, causing incorrect value restoration and dependency evaluation.

## 2. What Changed (Where)

| File | Change |
|------|--------|
| `prop-dependency-utils.ts` (editing-panel) | Refactored `updateValue()` to recursively traverse nested structures; added compatibility check for overridable wrapper consistency |
| `prop-dependency-utils.ts` (editor-props) | Enhanced `extractValue()` with `unwrapOverridableLeaf` option; added overridable traversal for nested paths |
| `is-overridable.ts` | New utility: type guards and rewrap function for overridable values |
| `index.ts` (editor-props) | Exported new overridable utilities and missing dependency-utils exports |
| Tests | Added 6 new test cases covering overridable restoration edge cases |

## 3. How It Works

When a dependency unmet condition triggers, the code now: (1) retrieves the saved value and current value with `unwrapOverridableLeaf: false` to preserve wrapper state, (2) checks compatibility via `isCompatibleSavedValue()` to ensure both are wrapped or both unwrapped, (3) uses `cloneDescent()` to recursively clone nested structures while maintaining overridable wrapper integrity, and (4) merges leaf values with `mergeLeafValue()` using `rewrapOverridableValue()` when needed to preserve override metadata.

## 4. Risks

**Shape drift scenarios**: If overridable wrappers are added/removed between save and restore, the restored value gracefully falls back to defaults rather than corrupting state. Mitigated by explicit compatibility check. No backwards compatibility issues since session storage is ephemeral.

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>

### CODE REVIEW

File: packages/packages/core/editor-editing-panel/src/utils/prop-dependency-utils.ts [hunk 1]
```diff
@@ -173,17 +176,58 @@ function updateValue( path: string[], value: Value, values: Values ) {
...
-			carry[ key ] = value ?? null;
+			const existing = carry[ key ];
+			carry[ key ] = mergeLeafValue( existing, value );
 
-			return ( carry[ key ]?.value as Values ) ?? carry.value;
+			return getDescentForPropUpdate( carry, key );
 		}
 
-		return ( carry[ key ]?.value as Values ) ?? carry.value;
+		return getDescentForPropUpdate( carry, key );
```
Comment (ronenboxer) [inline packages/packages/core/editor-editing-panel/src/utils/prop-dependency-utils.ts]: What about. this?
```suggestion
			const existing = carry[ key ];
			carry[ key ] = mergeLeafValue( existing, value );
}

		return getDescentForPropUpdate( carry, key );
```

File: packages/packages/core/editor-editing-panel/src/utils/prop-dependency-utils.ts [hunk 2]
```diff
@@ -173,17 +176,58 @@ function updateValue( path: string[], value: Value, values: Values ) {
...
 	}, newValue );
 
 	return { [ topPropKey ]: newValue[ topPropKey ] ?? null };
 }
 
+function getDescentForPropUpdate( carry: Values | null, key: string ) {
```
Comment (ronenboxer) [inline packages/packages/core/editor-editing-panel/src/utils/prop-dependency-utils.ts]: I'm not sure about the name
This function gets values for update, right?
If so, then maybe something like `getValueForPropUpdate`

File: packages/packages/libs/editor-props/src/utils/prop-dependency-utils.ts [hunk 3]
```diff
@@ -113,24 +114,47 @@ function getRelationMethod( relation: Relation ) {
...
+			if ( isOverridable( value ) ) {
+				const inner = value.value.origin_value;
+				value = isTransformable( inner ) ? inner.value ?? null : inner;
+			} else if ( isTransformable( value ) ) {
+				value = value.value ?? null;
+			}
+		}
 
-		return index !== path.length - 1 && isTransformable( value ) ? value.value ?? null : value;
+		return value;
```
Comment (ronenboxer) [inline packages/packages/libs/editor-props/src/utils/prop-dependency-utils.ts]: Just to reduce scopes and indentations
```suggestion
		if ( isLast ) {
			return value;
		}

		if ( isOverridable( value ) ) {
			const inner = value.value.origin_value;
			value = isTransformable( inner ) ? inner.value ?? null : inner;
		} else if ( isTransformable( value ) ) {
			value = value.value ?? null;
		}

		return value;
```

File: packages/packages/libs/editor-props/src/utils/__tests__/prop-dependency-utils.test.ts [hunk 4]
```diff
@@ -946,5 +946,78 @@ describe( 'prop-dependency-utils', () => {
 
 			expect( result ).toEqual( values );
 		} );
+
```
Comment (ronenboxer) [inline packages/packages/libs/editor-props/src/utils/__tests__/prop-dependency-utils.test.ts:949]: Maybe add more tests to cover combinations of overridable vs non-overridable with isDependencyMet?

**Review by ronenelementor — APPROVED** [review]

File: packages/packages/libs/editor-props/src/utils/__tests__/prop-dependency-utils.test.ts [hunk 5]
```diff
@@ -946,5 +946,115 @@ describe( 'prop-dependency-utils', () => {
...
+				},
+			};
+
+			const result = extractValue( [ 'block', 'inner' ], values );
+
+			expect( result ).toEqual( { $$type: 'boolean', value: true } );
+		} );
+	} );
+
+	describe( 'isDependencyMet with overridable affecting props', () => {
```
Comment (MiryamOren) [inline packages/packages/libs/editor-props/src/utils/__tests__/prop-dependency-utils.test.ts:1003]: Great 🤝 ⚡

**Review by MiryamOren — APPROVED** [review]
